File: MotivApp/app/src/main/java/com/motiv/piotr/RateLimit.java
package com.motiv.piotr;

import androidx.annotation.NonNull;
import androidx.room.*;
import com.google.gson.*;
import com.google.gson.annotations.*;
import com.google.gson.reflect.*;
import java.util.*;
import java.util.concurrent.*;

@Entity(tableName = "ratelimit")
public class RateLimit {

    private static final Gson gson = new Gson();
    @NonNull @PrimaryKey private java.lang.String id = UUID.randomUUID().toString();

    @ColumnInfo(name = "limit")
    @SerializedName("limit")
    private int limit;

    @ColumnInfo(name = "reset")
    @SerializedName("reset")
    private int reset;

    @ColumnInfo(name = "remaining")
    @SerializedName("remaining")
    private int remaining;

    public java.lang.String getId() {

        return this.id;
    }

    public void setId(java.lang.String id) {
        this.id = id;
    }

    public int getLimit() {

        return this.limit;
    }

    public void setLimit(int limit) {
        this.limit = limit;
    }

    public int getReset() {

        return this.reset;
    }

    public void setReset(int reset) {
        this.reset = reset;
    }

    public int getRemaining() {

        return this.remaining;
    }

    public void setRemaining(int remaining) {
        this.remaining = remaining;
    }

    public static RateLimit fromJson(String json) {
        return gson.fromJson(json, RateLimit.class);
    }

    public static String toJson(RateLimit object) {
        return gson.toJson(object, RateLimit.class);
    }

    public static String toJsonArray(RateLimit[] array) {
        return gson.toJson(array, RateLimit[].class);
    }

    public static RateLimit[] fromJsonArray(String array) {
        return gson.fromJson(array, RateLimit[].class);
    }
}

File: MotivApp/app/src/main/java/com/motiv/piotr/RateLimit.kt
 
package com.motiv.piotr
import android.widget.ImageView
import androidx.annotation.NonNull
import androidx.databinding.BindingAdapter
import com.google.gson.*
import com.google.gson.annotations.*
import com.google.gson.reflect.*
import com.squareup.picasso.Picasso
import io.realm.*
import java.util.*
import java.util.concurrent.*

public open class RateLimit : RealmObject() {

    @NonNull private var id: String = UUID.randomUUID().toString()

    @SerializedName("limit")

    private var limit: Int = 0

    @SerializedName("reset")

    private var reset: Int = 0

    @SerializedName("remaining")

    private var remaining: Int = 0

    fun getId(): String {
        return this.id
    } fun setId(id: String) {
        this.id = id
    } fun getLimit(): Int {
        return this.limit
    } fun setLimit(limit: Int) {
        this.limit = limit
    } fun getReset(): Int {
        return this.reset
    } fun setReset(reset: Int) {
        this.reset = reset
    } fun getRemaining(): Int {
        return this.remaining
    } fun setRemaining(remaining: Int) {
        this.remaining = remaining
    } @BindingAdapter("bind:imageUrl")
    fun loadImage(view: ImageView, url: String) {
        Picasso.with(view.getContext()).load(url).into(view)
    } companion object {
        val gson: Gson = Gson()
        fun fromJson(json: String): RateLimit {
            return gson.fromJson(json, RateLimit::class.java)
        }

        fun toJson(obj: RateLimit): String {
            return gson.toJson(obj, RateLimit::class.java)
        }

        fun toJsonArray(obj: RateLimit): String {
            return gson.toJson(obj, Array<RateLimit>::class.java)
        }

        fun fromJsonArray(json: String): Array<RateLimit> {
            return gson.fromJson(json, Array<RateLimit>::class.java)
        }
    }
}
